src/Components/DrillPage.js

```diff
@@ -1,62 +1,79 @@
-import React, { Component } from 'react';
-import { ScrollView, StyleSheet, View, Text, Button, ImageBackground, TouchableOpacity } from 'react-native';
+import React from 'react';
+import { ScrollView, StyleSheet, View, Text, Image, ImageBackground, TouchableOpacity } from 'react-native';
 import { connect } from 'react-redux';
+import ic_favorite_border from '../Images/ic_favorite_border.png';
+import ic_favorite from '../Images/ic_favorite.png';
 import theme from '../styles/theme.style';
 
-class DrillPage extends Component {
-  _toggleFavorite() {
-    const action = { type: 'TOGGLE_FAVORITE', value: this.state.dr };
-    this.props.dispatch(action);
-  }
-
-  componentDidUpdate() {
-    console.log('componentDidUpdate : ');
-    console.log(this.props.favoritesDrill);
+class DrillPage extends React.Component {
+  state = {
+    drill: this.props.drill,
+  };
+  _displayFavoriteImage() {
+    var sourceImage = ic_favorite_border;
+    if (this.props.favoritesDrill.findIndex(item => item.id === this.props.route.params.drill.id) !== -1) {
+      sourceImage = ic_favorite;
+    }
+    return <Image style={styles.favoriteImage} source={sourceImage} />;
   }
 
   render() {
     const drill = this.props.route.params.drill;
     const { navigation } = this.props;
+    const {
+      title,
+      infoWrapper,
+      DrillPage,
+      separator,
+      videoLinkText,
+      favoriteContainer,
+      image,
+      imageOpacity,
+      info,
+      videoLink,
+      description,
+      descriptionItem,
+      descriptionText,
+      descriptionTitle,
+    } = styles;
 
     return (
-      <ScrollView style={styles.DrillPage}>
-        <ImageBackground source={{ uri: drill.image }} style={styles.image} imageStyle={styles.imageOpacity}>
-          <Text style={styles.title}>{drill.title}</Text>
-          <View style={styles.infoWrapper}>
-            <Text style={styles.info}>{drill.durationInMinutes} minutes</Text>
-            <View style={styles.separator} />
-            <Text style={styles.info}>{drill.nbPlayers} players</Text>
-            <View style={styles.separator} />
-            <Text style={styles.info}>{drill.level} level</Text>
+      <ScrollView style={DrillPage}>
```

Comment:
> I'm not sure about styles destructuration
> 
> Discussing it there: [#54 (comment)](https://github.com/UltimateApp/UltimateApp/pull/54#discussion_r403677912)

This is exactly the kind of problems we are going to run into if we destructure styles. We can't see the difference anymore between objects useful for the components behaviour (such as extracted from props, state, imported from other modules) from style objects

I feel like we have thrown away clarity and it makes the component much more difficult to read.
Would be nice to have more advices on the topic before merging this